code/modules/mapping/mapping_helpers.dm

```diff
@@ -446,28 +446,84 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	name = "Dead Body placer"
 	late = TRUE
 	icon_state = "deadbodyplacer"
+	var/admin_spawned
 	var/bodycount = 2 //number of bodies to spawn
 
+/obj/effect/mapping_helpers/dead_body_placer/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		return
+	admin_spawned = TRUE
+
 /obj/effect/mapping_helpers/dead_body_placer/LateInitialize()
 	var/area/a = get_area(src)
 	var/list/trays = list()
 	for (var/i in a.contents)
 		if (istype(i, /obj/structure/bodycontainer/morgue))
+			if(admin_spawned)
+				var/obj/structure/bodycontainer/morgue/early_morgue_tray = i
+				if(early_morgue_tray.connected.loc != early_morgue_tray)
+					continue
 			trays += i
 	if(!trays.len)
-		log_mapping("[src] at [x],[y] could not find any morgues.")
+		if(admin_spawned)
+			message_admins("[src] spawned at [ADMIN_VERBOSEJMP(src)] failed to find a closed morgue to spawn a body!")
+		else
+			log_mapping("[src] at [x],[y] could not find any morgues.")
 		return
+
+	var/reuse_trays = (trays.len < bodycount) //are we going to spawn more trays than bodies?
+
+	var/use_species = CONFIG_GET(flag/morgue_cadaver_disable_nonhumans)
+	var/species_probability = CONFIG_GET(number/morgue_cadaver_other_species_probability)
+	var/override_species = CONFIG_GET(string/morgue_cadaver_override_species)
+	var/list/usable_races
+	if(use_species)
+		usable_races = GLOB.roundstart_races.Copy()
+		usable_races -= SPECIES_ETHEREAL //they revive on death which is bad juju
```

Comment:
I really dont like hardcoding stuff like this.

Reply:
sadly this is one extremely snowflake where they're literally litches that have their bodies disappear and revive on death, so there's not a way around this.